src/System.IO.Pipelines.Networking.Tls/ISecureContext.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Threading.Tasks;
+
+namespace System.IO.Pipelines.Networking.Tls
+{
+    public interface ISecureContext : IDisposable
+    {
+        int TrailerSize { get; set; }
+        int HeaderSize { get; set; }
+        bool ReadyToSend { get; }
+        ApplicationProtocols.ProtocolIds NegotiatedProtocol { get; }
+        Task ProcessContextMessageAsync(ReadableBuffer readBuffer, IPipelineWriter writer);
```

Comment:
It's pretty unclear (to me) what this should do.  Since there's nothing passed in indicating what context the caller is assuming.